### PR TITLE
Fix flaky end to end tests

### PR DESCRIPTION
**Pull request by pengchen0692**
## Description
Randomize instance identifier of test files, so that never get conflict error  when upload.

## Related issues
Addresses [AB#80262](https://microsofthealth.visualstudio.com/f8da5110-49b1-4e9f-9022-2f58b6124ff9/_workitems/edit/80262)

## Testing
Locally passed.

### CODE REVIEW

File: src/Microsoft.Health.Dicom.Tests.Common/Comparers/DicomFileEqualityComparer.cs [hunk 1]
```diff
@@ -5,31 +5,45 @@
...
 using System.Collections.Generic;
 using Dicom;
+using EnsureThat;
 
 namespace Microsoft.Health.Dicom.Tests.Common.Comparers
 {
     public class DicomFileEqualityComparer : IEqualityComparer<DicomFile>
     {
+        public DicomFileEqualityComparer()
+           : this(new DicomTag[0])
```
Comment (wsugarman) [inline src/Microsoft.Health.Dicom.Tests.Common/Comparers/DicomFileEqualityComparer.cs:15]: nit: `Array.Empty<DicomTag>`

File: src/Microsoft.Health.Dicom.Tests.Common/Comparers/DicomFileEqualityComparer.cs [hunk 2]
```diff
@@ -5,31 +5,45 @@
...
+        {
+        }
+
+        public DicomFileEqualityComparer(IEnumerable<DicomTag> ignoredTags)
+        {
+            EnsureArg.IsNotNull(ignoredTags, nameof(ignoredTags));
+            IgnoredTags = ignoredTags;
+        }
+
         bool IEqualityComparer<DicomFile>.Equals(DicomFile x, DicomFile y)
```
Comment (wsugarman) [inline src/Microsoft.Health.Dicom.Tests.Common/Comparers/DicomFileEqualityComparer.cs:25]: Is there a particular reason we're explicitly implementing?

File: src/Microsoft.Health.Dicom.Tests.Common/Comparers/DicomFileEqualityComparer.cs [hunk 3]
```diff
@@ -5,31 +5,45 @@
...
         {
             if (x == null || y == null)
             {
                 return object.ReferenceEquals(x, y);
             }
 
-            IEqualityComparer<IEnumerable<DicomItem>> metadataComparer = new DicomItemCollectionEqualityComparer();
+            IEqualityComparer<IEnumerable<DicomItem>> metadataComparer = new DicomItemCollectionEqualityComparer(IgnoredTags);
```
Comment (wsugarman) [inline src/Microsoft.Health.Dicom.Tests.Common/Comparers/DicomFileEqualityComparer.cs:32]: Can we create this comparer once?

File: src/Microsoft.Health.Dicom.Tests.Common/Comparers/DicomFileEqualityComparer.cs [hunk 4]
```diff
@@ -5,31 +5,45 @@
...
+            IEqualityComparer<DicomDataset> dataSetComparer = new DicomDatasetEqualityComparer(IgnoredTags);
             return dataSetComparer.Equals(x.Dataset, y.Dataset);
         }
 
         int IEqualityComparer<DicomFile>.GetHashCode(DicomFile obj)
         {
             return obj.GetHashCode();
         }
+
+        public IEnumerable<DicomTag> IgnoredTags { get; }
```
Comment (wsugarman) [inline src/Microsoft.Health.Dicom.Tests.Common/Comparers/DicomFileEqualityComparer.cs:47]: Can you move this to the top?

Comment (wsugarman) [inline src/Microsoft.Health.Dicom.Tests.Common/Comparers/DicomFileEqualityComparer.cs:47]: Does this need to be a `public` property?

File: test/Microsoft.Health.Dicom.Web.Tests.E2E/Rest/RetrieveTransactionResourceTests.Frame.cs [hunk 5]
```diff
@@ -60,6 +60,7 @@ await foreach (Stream item in response)
                 Assert.Equal(item.ToByteArray(), pixelData.GetFrame(frameIndex).Data);
                 frameIndex++;
             }
+
```
Comment (wsugarman) [inline test/Microsoft.Health.Dicom.Web.Tests.E2E/Rest/RetrieveTransactionResourceTests.Frame.cs:63]: nit: extra newline

File: test/Microsoft.Health.Dicom.Web.Tests.E2E/Rest/RetrieveTransactionResourceTests.Instance.cs [hunk 6]
```diff
@@ -32,16 +31,25 @@ public async Task GivenSupportedAcceptHeaders_WhenRetrieveInstance_ThenServerSho
...
+            await InternalStoreAsync(new[] { inputDicomFile });
 
             using DicomWebResponse<DicomFile> response = await _client.RetrieveInstanceAsync(instanceId.StudyInstanceUid, instanceId.SeriesInstanceUid, instanceId.SopInstanceUid, transferSyntax);
 
             Assert.Equal(DicomWebConstants.ApplicationDicomMediaType, response.ContentHeaders.ContentType.MediaType);
 
-            DicomFile actual = await response.GetValueAsync();
-            DicomFile expected = DicomFile.Open(transcoderTestData.ExpectedOutputDicomFile);
-            Assert.Equal(expected, actual, new DicomFileEqualityComparer());
+            var actual = (await response.GetValueAsync());
```
Comment (wsugarman) [inline test/Microsoft.Health.Dicom.Web.Tests.E2E/Rest/RetrieveTransactionResourceTests.Instance.cs:42]: nit: unnecessary `()`

**Review by wsugarman — APPROVED** [review]

File: src/Microsoft.Health.Dicom.Tests.Common/Comparers/DicomFileEqualityComparer.cs (hunk 2 above)
Comment (pengchen0692) [reply src/Microsoft.Health.Dicom.Tests.Common/Comparers/DicomFileEqualityComparer.cs:25]: no. will fix it.

**Pull request merged**